frontend/src/features/accounts/org-form.ts

```diff
@@ -0,0 +1,187 @@
+            label=${msg("Org name")}
+            placeholder=${msg("My Organization")}
+            autocomplete="off"
+            value=${this.name === this.orgId ? "" : this.name}
+            minlength="2"
+            maxlength="40"
+            help-text=${msg("You can change this in your org settings later.")}
+            required
+          ></sl-input>
+        </div>
+        <div class="mb-5">
+          <sl-input
+            name="orgSlug"
+            label=${msg("Custom URL identifier")}
+            placeholder="my-organization"
+            autocomplete="off"
+            value=${this.slug}
+            minlength="2"
+            maxlength="30"
+            help-text=${helpText(this.slug)}
+            required
+            @sl-input=${(e: InputEvent) => {
+              const input = e.target as SlInput;
+              input.helpText = helpText(slugify(input.value, { strict: true }));
+            }}
+          >
+          </sl-input>
+        </div>
+        ${this.renameOrgTask.render({
+          error: (err) =>
+            html`<div class="my-5">
+              <btrix-alert id="formError" variant="danger"
+                >${err instanceof Error ? err.message : err}</btrix-alert
+              >
+            </div>`,
+        })}
+        <sl-button
+          class="w-full"
+          variant="primary"
+          type="submit"
+          ?loading=${this.renameOrgTask.status === TaskStatus.PENDING}
+        >
+          ${msg("Go to Dashboard")}
+        </sl-button>
+      </form>
+    `;
+  }
+
+  private async onSubmit(e: SubmitEvent) {
+    e.preventDefault();
+
+    const form = e.target as HTMLFormElement;
+    if (!(await this.checkFormValidity(form))) return;
+
+    const params = serialize(form) as OrgFormSubmitEventDetail["values"];
+    const orgName = params.orgName;
+    const orgSlug = slugify(params.orgSlug);
+
+    void this.renameOrgTask.run([this.orgId, orgName, orgSlug]);
+  }
+
+  private async renameOrg(
+    id: string,
+    params: { name?: string; slug?: string },
+  ) {
+    const name = params.name || this.name;
+    const slug = params.slug || this.slug;
+
+    try {
+      await this.api.fetch(`/orgs/${id}/rename`, this.authState!, {
+        method: "POST",
+        body: JSON.stringify({ name, slug }),
+      });
+      this.notify.toast({
+        message: msg("Org successfully updated."),
+        variant: "success",
+        icon: "check2-circle",
+      });
+
+      this.dispatchEvent(
+        new CustomEvent<UpdateUserInfoDetail>("btrix-update-user-info", {
+          detail: {
+            updateComplete: () => {
+              this.goToOrgDashboard(slug);
+            },
+          },
+          bubbles: true,
+        }),
+      );
+    } catch (e) {
+      console.debug(e);
+      if (isApiError(e) && e.details === "duplicate_org_name") {
+        throw new Error(
+          msg("This org name or URL is already taken, try another one."),
+        );
+      }
```

Comment:
Once https://github.com/webrecorder/browsertrix/pull/1891 is merged, we can also check for an invalid slug, which will return the detail `invalid_slug`).

If we want to check that an org slug is not a duplicate of an existing one ahead of time, we also have the `GET /orgs/slugs` and `GET /orgs/slug-lookup` endpoints available already in the API.

Reply:
Moved to issue here: https://github.com/webrecorder/browsertrix/issues/1892